src/framework/auth/strategies/default/default-strategy-options.ts

```diff
@@ -18,30 +22,89 @@ export interface NbDefaultStrategyModule {
   defaultMessages?: string[];
 }
 
-// TODO postfix to options
 export interface NbDefaultStrategyReset extends NbDefaultStrategyModule {
   resetPasswordTokenKey?: string;
 }
 
-// TODO postfix to options
-export interface NbDefaultAuthStrategyOptions {
-  baseEndpoint?: string;
-  login?: boolean | NbDefaultStrategyModule;
-  register?: boolean | NbDefaultStrategyModule;
-  requestPass?: boolean | NbDefaultStrategyModule;
-  resetPass?: boolean | NbDefaultStrategyReset;
-  logout?: boolean | NbDefaultStrategyReset;
-  token?: {
-    key?: string;
-    getter?: Function;
+export class NbDefaultAuthStrategyOptions extends NbAuthStrategyOptions {
+  name = 'email';
+  baseEndpoint? = '/api/auth/';
+  login?: boolean | NbDefaultStrategyModule = {
+    alwaysFail: false,
+    rememberMe: true, // TODO: what does that mean?
+    endpoint: 'login',
+    method: 'post',
+    redirect: {
+      success: '/',
+      failure: null,
+    },
+    defaultErrors: ['Login/Email combination is not correct, please try again.'],
+    defaultMessages: ['You have been successfully logged in.'],
+  };
+  register?: boolean | NbDefaultStrategyModule = {
+    alwaysFail: false,
+    rememberMe: true,
+    endpoint: 'register',
+    method: 'post',
+    redirect: {
+      success: '/',
+      failure: null,
+    },
+    defaultErrors: ['Something went wrong, please try again.'],
+    defaultMessages: ['You have been successfully registered.'],
+  };
+  requestPass?: boolean | NbDefaultStrategyModule = {
+    endpoint: 'request-pass',
+    method: 'post',
+    redirect: {
+      success: '/',
+      failure: null,
+    },
+    defaultErrors: ['Something went wrong, please try again.'],
+    defaultMessages: ['Reset password instructions have been sent to your email.'],
+  };
+  resetPass?: boolean | NbDefaultStrategyReset = {
+    endpoint: 'reset-pass',
+    method: 'put',
+    redirect: {
+      success: '/',
+      failure: null,
+    },
+    resetPasswordTokenKey: 'reset_password_token',
+    defaultErrors: ['Something went wrong, please try again.'],
+    defaultMessages: ['Your password has been successfully changed.'],
   };
-  errors?: {
-    key?: string;
-    getter?: Function;
+  logout?: boolean | NbDefaultStrategyReset = {
+    alwaysFail: false,
+    endpoint: 'logout',
+    method: 'delete',
+    redirect: {
+      success: '/',
+      failure: null,
+    },
+    defaultErrors: ['Something went wrong, please try again.'],
+    defaultMessages: ['You have been successfully logged out.'],
   };
-  messages?: {
-    key?: string;
-    getter?: Function;
+  token = {
+    key: 'data.token',
+    getter: (module: string, res: HttpResponse<Object>, options: any) => getDeepFromObject(res.body,
```

Comment:
Also, why do we use *options: **any*** in all getters?

Reply:
as it's getting to long :)